actors/builtin/market/market_balances.go

```diff
@@ -104,6 +104,36 @@ func (m *marketStateMutation) maybeLockBalance(addr addr.Address, amount abi.Tok
 	return nil, exitcode.Ok
 }
 
+func (m *marketStateMutation) removeAccountIfNoBalance(addr addr.Address) (error, exitcode.ExitCode) {
+	bal, err, code := getBalance(m.escrowTable, addr)
+	if err != nil {
+		return xerrors.Errorf("failed to get escrow balance: %w", err), code
+	}
+
+	if bal.Equals(big.Zero()) {
+		prev, err := m.escrowTable.Remove(addr)
+		if err != nil {
+			return xerrors.Errorf("failed to remove account: %w", err), exitcode.ErrIllegalState
+		}
+		AssertMsg(prev.Equals(big.Zero()), "previous escrow balance should be zero")
+
+		// remove locked balance account
+		bal, err, code = getBalance(m.lockedTable, addr)
```

Comment:
No need to nest this inside the outer `if`.

Reply:
@anorth It is possible for locked bal to be zero without the escrow bal being zero in which case we don't want to remove the locked account just yet.